src/security/CertGadgets.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 1996-2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_SECURITY_CERTGADGETS_H
+#define SQUID_SRC_SECURITY_CERTGADGETS_H
+
+#include "security/forward.h"
+
+namespace Security
+{
+
+/// \return the SubjectName field of the given certificate
+SBuf CertSubjectName(const CertPointer &);
+
+/// \return the Issuer field of the given certificate
+SBuf CertIssuerName(const CertPointer &);
+
+/// \returns whether cert was (correctly) issued by the given issuer
+/// Due to complexity of the underlying checks, it is impossible to clearly
+/// distinguish pure negative answers (e.g., two independent certificates)
+/// from errors (e.g., the issuer certificate lacks the right CA extension).
+bool CertIsIssuedBy(const CertPointer &cert, const CertPointer &issuer);
+
+/// convenience wrapper for checking self-signed certificates
+inline bool CertIsSelfSigned(const CertPointer &cert) {
```

Comment:
IIRC, we do not use `inline` for explicitly/immediately inlined stuff anymore:
```suggestion
bool CertIsSelfSigned(const CertPointer &cert) {
```

Reply:
I don't recall making that change for functions in .h files. You may be mixing it up with method definitions no longer being inline.

Reply:
AFAICT, C++ still requires explicit `inline` for non-member functions defined entirely in headers where they are declared. Sorry I have missed or forgotten that fact. Resolving.